### PR TITLE
Analytics component

### PR DESCRIPTION
**Pull request by JessicaTXu**

### CODE REVIEW

File: src/Analytics.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+function trackAnalytics(eventType, businessId, siteId) {
+    var jsonData = {};
+    var pagesReferrer = window.document.referrer;
+    var pageurl = window.location.pathname;
+
+    jsonData['pagesReferrer'] = pagesReferrer;
+    jsonData['pageurl'] = pageurl;
+    jsonData['eventType'] = eventType;
+    jsonData['businessids'] = businessId;
```
Comment (bmayyext) [inline src/Analytics.js:25]: is the key supposed to be 'businessids' or 'businessId'?

File: partials/analytics.hbs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    function trackAnalytics(eventType, businessId, siteId) {
+        var jsonData = {};
+        var pagesReferrer = window.document.referrer;
+        var pageurl = window.location.pathname;
+
+        jsonData['pagesReferrer'] = pagesReferrer;
+        jsonData['pageurl'] = pageurl;
+        jsonData['eventType'] = eventType;
+        jsonData['businessids'] = businessId;
+        jsonData['siteId'] = siteId;
```
Comment (bmayyext) [inline partials/analytics.hbs:28]: var jsonData = {
  'pagesReferrer': window.document.referrer,
  'pageurl': window.location.pathname,
  'eventType': eventType,
  'businessids': businessId,
  'siteId': siteId,
};

File: partials/analytics.hbs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    function trackAnalytics(eventType, businessId, siteId) {
+        var jsonData = {};
+        var pagesReferrer = window.document.referrer;
+        var pageurl = window.location.pathname;
+
+        jsonData['pagesReferrer'] = pagesReferrer;
+        jsonData['pageurl'] = pageurl;
+        jsonData['eventType'] = eventType;
+        jsonData['businessids'] = businessId;
```
Comment (bmayyext) [inline partials/analytics.hbs:27]: is the key supposed to be 'businessids' or 'businessId'?

File: partials/analytics.hbs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{{!-- <script src  ="Analytics.js"></script> --}}
+<script>
+    function seed() {
+        return Date.now() + Math.floor(1000 * Math.random());
+    }
+
+    function pixelURL(data) {
+    data.v = seed();
+    let p = '//www.yext-pixel.com/store_pagespixel?product=sites';
```
Comment (bmayyext) [inline partials/analytics.hbs:9]: `const`, not `let`

File: templates/entity-page.hbs [hunk 5]
```diff
@@ -6,7 +6,7 @@
         "destination": "pages",
         "filter" : {
             "entityTypes": [
-                "[REPLACE ME]"
+                "ce_summerOutfit"
```
Comment (bmayyext) [inline templates/entity-page.hbs:9]: this doesn't seem like it should have changed

File: partials/analytics.hbs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    for (var key in data) {
+        if (data.hasOwnProperty(key)) {
+        p += `&${key}=${encodeURIComponent(data[key])}`;
+        }
+    }
+
+    return p;
```
Comment (bmayyext) [inline partials/analytics.hbs:16]: indent right 2 spaces

File: sites-global/global.json [hunk 7]
```diff
@@ -1,6 +1,7 @@
 {
     "name": "Yext's Starter",
-    "accountId": "",
+    "accountId": "1234",
```
Comment (bmayyext) [inline sites-global/global.json:3]: was this change supposed to be made?

File: sites-global/global.json [hunk 8]
```diff
@@ -1,6 +1,7 @@
...
     "googleApiKey": "",
-    "liveApiKey": ""
+    "liveApiKey": "",
+    "siteId": "1234"
```
Comment (bmayyext) [inline sites-global/global.json:6]: based on other values, should this be empty as well?

**Review by bmayyext — CHANGES_REQUESTED** [review]
Indent size is generally inconsistent (2 spaces on some lines, 4 spaces on others).

File: src/Analytics.js (hunk 1 above)
Comment (JessicaTXu) [reply src/Analytics.js:25]: businessids is what's expected

File: sites-global/global.json (hunk 7 above)
Comment (JessicaTXu) [reply sites-global/global.json:3]: Yeah, this is just something I'm using in lieu of https://yexttest.atlassian.net/browse/SPR-4278. Will undo this change once the massaged data is available.

**Comment by JessicaTXu** [comment]
Looks like analytics.js is not longer the strictmode, it now includes the pixel stuff, but I'm still getting Uncaught ReferenceError: trackAnalytics is not defined

File: src/Analytics.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  return p;
+}
+
+function trackAnalytics(eventType, businessId, siteId) {
+    var jsonData = {
+      'pagesReferrer': window.document.referrer,
+      'pageurl': window.location.pathname,
+      'eventType': eventType,
+      'businessids': businessId,
+      'siteId': siteId,
```
Comment (bmayyext) [inline src/Analytics.js:23]: minor, but use 2 space indents

File: src/Analytics.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
+function seed() {
+  return Date.now() + Math.floor(1000 * Math.random());
+}
+
+function pixelURL(data) {
+  data.v = seed();
+  let p = '//www.yext-pixel.com/store_pagespixel?product=sites';
+  for (var key in data) {
+    if (data.hasOwnProperty(key)) {
```
Comment (bmayyext) [inline src/Analytics.js:9]: should we also check that the data value is not empty, i.e.

`if (data.hasOwnProperty(key) && !!data[key]) {`

**Review by bmayyext — COMMENTED** [review]
We should probably squash this into one commit before merging, right?

**Comment by mkilpatrick** [comment]
LGTM

**Review by bmayyext — APPROVED** [review]

**Pull request merged**